packages/@aws-cdk/aws-cloudwatch/lib/alarm.ts

```diff
@@ -120,6 +220,10 @@ export class Alarm extends Resource implements IAlarm {
     class Import extends Resource implements IAlarm {
       public readonly alarmArn = alarmArn;
       public readonly alarmName = Stack.of(scope).parseArn(alarmArn, ':').resourceName!;
+
+      public toAlarmRule(): string {
+        throw new Error('Method not implemented.');
```

Comment:
Well why not though?

Change this to `extends AlarmBase` (instead of `extends Resource`) and you get a working implementation for free.

(Good call that refactor)

Reply:
that's a good point, my thought process was driven by making the state configurable, but if we're grounding it to ```ALARM```, this becomes a simpler less controversial implementation.